Author: Stefan Seyfried <seife+obs@b1-systems.com>

This patch fixes only the most disgusting compiler errors
and 64 bit issues, all caused by missing includes or function
prototypes.
It is not pretty.


Index: b/board.c
===================================================================
--- a/board.c
+++ b/board.c
@@ -1,9 +1,13 @@
 #include "board.h"
 
 #include <stdio.h>
 
+#include "ticks.h"
+
+extern void sndPlayOnce(int sample, int posX);
+
 
 static int isWall(playField* pf, int x, int y)
 {
   if(x < 0) return(0);
   if(x+1 > FIELDSIZE) return(0);
Index: b/dumplevelimages.c
===================================================================
--- a/dumplevelimages.c
+++ b/dumplevelimages.c
@@ -3,10 +3,12 @@
 #include "board.h"
 #include "draw.h"
 #include "cursor.h"
 #include "player.h"
 
+extern void drawDisableCursor(void);
+
 void dumplevelimages(SDL_Surface* screen)
 {
   //Loop through levels
   int l;
   playField pf;
Index: b/menu.h
===================================================================
--- a/menu.h
+++ b/menu.h
@@ -10,9 +10,9 @@
 #include "stars.h"
 #include "leveleditor.h"
 #include "particles.h"
 
 int initMenu(); //Loads menu graphics
-int runMenu();
+int runMenu(SDL_Surface* screen);
 void setMenu(int mstate);
 
 #endif // MENU_H_INCLUDED
Index: b/menu.c
===================================================================
--- a/menu.c
+++ b/menu.c
@@ -5,10 +5,17 @@
 
 #include "states.h"
 #include "game.h"
 #include "levelselector.h"
 
+
+#include <SDL/SDL_image.h>
+
+extern void sndPlayOnce(int sample, int posX);
+extern void txtWave( SDL_Surface* scr, int fontNum, const char* txt, int x, int y, float* rot);
+extern void getUserLevelDir( char* dir );
+
 static float rot=0;
 static float rot2=0;
 static float rot3=0;
 static float rot4=0;
 
Index: b/sprite.c
===================================================================
--- a/sprite.c
+++ b/sprite.c
@@ -1,7 +1,9 @@
 #include "sprite.h"
 
+#include <SDL/SDL_image.h>
+
 SDL_Surface* loadImg( const char* fileName )
 {
   //Load the surface
   SDL_Surface* unoptimized = NULL;
   SDL_Surface* optimized = NULL;
Index: b/game.c
===================================================================
--- a/game.c
+++ b/game.c
@@ -1,7 +1,8 @@
 #include "game.h"
 
+extern void sndPlayOnce(int sample, int posX);
 
 static playField pf;
 static cursorType cur;
 static int countdown;
 static int countdownSeconds;
Index: b/player.c
===================================================================
--- a/player.c
+++ b/player.c
@@ -1,16 +1,18 @@
 #include "player.h"
+#include "levels.h"
 
 #include <stdio.h>
 
 #ifdef GP2X
 #include "wiz.h"
 #endif
 
 #ifdef LINUX
 #include <sys/stat.h>
 #include <string.h>
+#include <stdlib.h>
 #endif
 
 static player_t playerInfo;
 
 player_t* player()
Index: b/levels.c
===================================================================
--- a/levels.c
+++ b/levels.c
@@ -7,10 +7,12 @@
 
 #ifdef LINUX
 #include <sys/stat.h>
 #endif
 
+extern void getUserLevelDir( char* dir );
+
 static listItem* levelFiles;
 static listItem* userLevelFiles;
 static int numUserLevels=0;
 static int numLevels=0;
 
Index: b/pixel.c
===================================================================
--- a/pixel.c
+++ b/pixel.c
@@ -1,7 +1,8 @@
 #include "pixel.h"
 #include "math.h"
+#include "ticks.h"
 
 inline void plotPixel(SDL_Surface* img, int x, int y, uint16_t col)
 {
   //Bail if invalid position
   if(x < 0 || x > 319 || y < 0 || y > 239)
Index: b/leveleditor.c
===================================================================
--- a/leveleditor.c
+++ b/leveleditor.c
@@ -7,10 +7,12 @@
 #include "text.h"
 #include "levels.h"
 #include "list.h"
 #include <stdio.h>
 
+extern void addUserLevel(const char* fn);
+
 static playField pf;
 static cursorType cur;
 static int selBrick=1;
 static char saveFile[32];
 static int changed=0;
Index: b/stars.c
===================================================================
--- a/stars.c
+++ b/stars.c
@@ -1,8 +1,9 @@
 #include "stars.h"
 #include "pixel.h"
 #include "sound.h"
+#include "ticks.h"
 
 static listItem* stars;
 static listItem* rockets;
 //Setup 1000 stars
 #define NUMSTARS 500
Index: b/main.c
===================================================================
--- a/main.c
+++ b/main.c
@@ -14,10 +14,12 @@
 #include "game.h"
 #include "levels.h"
 #include "leveleditor.h"
 #include "particles.h"
 
+#include <SDL/SDL_image.h>
+
 #ifndef GP2X
   #include "dumplevelimages.h"
 #endif
 
 int main(int argc, char *argv[])
